.pre-commit-config.yaml

```diff
@@ -20,29 +20,31 @@ repos:
     -   id: end-of-file-fixer
 
 -   repo: https://github.com/ambv/black
-    rev: 19.10b0
+    rev: 21.10b0
     hooks:
     - id: black
 
 -   repo: https://gitlab.com/pycqa/flake8
-    rev: 3.7.9
+    rev: 4.0.1
     hooks:
     -   id: flake8
+        args: [--show-source, --statistics]
 
 -   repo: https://github.com/asottile/seed-isort-config
-    rev: v2.1.0
+    rev: v2.2.0
     hooks:
     -   id: seed-isort-config
 
 -   repo: https://github.com/pycqa/isort
-    rev: 5.6.4
+    rev: 5.10.1
     hooks:
     -   id: isort
         exclude: README.md
         additional_dependencies: [toml]
 
 -   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: 'v0.790'
+    rev: 'v0.910'
```

Comment:
How do we ensure that we keep updating this config regularly?

Reply:
We should have a task in our quarterly fix-it log to ensure that the CI system is working as expected. Based on my limited usage so far there are two tasks which should be done every quarter (and these count towards Better Engg as well):
1. Update the version of all the linters (black, mypy, isort, flake8 etc.)
2. Revisit tests to ensure that they are evenly split by time i.e. all three partitions take approximately the same amount of time and perhaps increase the parallelism, if needed. Here's a link which shows how to achieve this. https://circleci.com/docs/2.0/parallelism-faster-jobs/#running-split-tests. The current state can be assessed from the attached image.

![circle_ci_timings](https://user-images.githubusercontent.com/91992329/141390199-c2a8546e-fa2c-431f-b67f-ff70a5e7c00b.png)